### PR TITLE
feature(wren-ai-service): support async SQL dry-run and async Qdrant component 

### PR DESCRIPTION
**Pull request by paopa**
This PR aims to make the SQL dry-run and Qdrant components support async tasks. and also extract the function related to the engine to a module.

- https://python-client.qdrant.tech/qdrant_client.async_qdrant_client
- https://docs.aiohttp.org/en/stable/client_quickstart.html#make-a-request

### CODE REVIEW

File: wren-ai-service/pyproject.toml [hunk 1]
```diff
@@ -21,6 +21,7 @@ sqlparse = "==0.5.0"
 sqlglot = "==22.5.0"
 orjson = "==3.10.3"
 sf-hamilton = {version = "1.63.0", extras = ["visualization"]}
+aiohttp = "3.9.5"
```
Comment (cyyeh) [inline wren-ai-service/pyproject.toml:24]: should be `aiohttp = "==3.9.5"`?

File: wren-ai-service/src/engine.py
Comment (cyyeh) [inline wren-ai-service/src/engine.py:1]: nice abstraction

**Review by cyyeh — COMMENTED** [review]
Overall LGTM, only minor issue

File: wren-ai-service/pyproject.toml (hunk 1 above)
Comment (cyyeh) [reply wren-ai-service/pyproject.toml:24]: also `sf-hamilton = {version = "==1.63.0", extras = ["visualization"]}`?

Comment (paopa) [reply wren-ai-service/pyproject.toml:24]: according to https://python-poetry.org/docs/dependency-specification/#exact-requirements, both of them are the way to specify the exact version for a package. I think we could unify using `==` to specify the version. I'll modify them in the next commit. 👍

**Review by cyyeh — APPROVED** [review]
LGTM

**Pull request merged**